### ISSUE TITLE
Admin login should not clear cache but ignore instead

### ISSUE THREAD

**Issue by abrookbanks** [cubecart/v6#1536]

**Comment by havenswift-hosting** [comment]
Any news on this issue ?

**Comment by abrookbanks** [comment]
I don't think a solution is possible until CubeCart v7.

**Comment by havenswift-hosting** [comment]
Is there any particular reason why ?  This particularly makes memcache and memcached effectively unusable as multiple sites on a single server using memcache continuously clearing cache creates race conditions in memcache which causes memcache to use huge amounts of memory and CPU when memcache should be very low footprint and highly effective.

This is on top of the general problem with all caching methods where the cache isnt very effective when an admin is in the store as it is continuously being cleared and created, cleared, created.  A good caching method should be 95% plus cache hits and CubeCart averages 35%

**Comment by abrookbanks** [comment]
Because there is no way to distinguish what cached elements are for what. We can't clear the cache for product id 5 for example. Ideally if product 5 is edited it clears the cache for just that ID. As CubeCart mainly caches SQL queries with an MD5 identifier we're stuffed.

**Comment by havenswift-hosting** [comment]
There are obviously ways around that although it is more work and not a quick fix.  However a quick solution (and the way other systems handle this situation - for example see http://docs.magento.com/m2/ce/user_guide/system/cache-management.html and WordPress cache plugins work in exactly the same way) is to disable all the points where cache is cleared in admin and then have a banner displayed in admin to show that cache has been invalidated and should be refreshed.  It is the choice of the admin user when they clear cache and can do it once when they have finished making their changes

**Comment by havenswift-hosting** [comment]
Doing more investigation and my suggestion above (banner notice plus admin clearing cache manually) seems to be the way that many other applications work with cache.  This would be a VAST improvement on how it currently works

**Comment by abrookbanks** [comment]
Sorry the milestone keeps moving on this. It's going to happen and I'm keen to get working on this.

**Comment by abrookbanks** [comment]
I think this is important but people are going to hate it.

**Comment by havenswift-hosting** [comment]
Sorry but I really dont understand why you think people will hate it - what makes you think this ?

It is common practice with most other systems as far as I can see - Magento certainly uses it and WordPress (that currently run approx 30% of all websites !!) also does it this way.  Most people dont understand that CubeCart currently clears cache with every single admin action making the cache almost useless (while an admin user is logged in) and those that find out about it are horrified that it works this way. I have used both Magento and many WordPress sites using this mechanism and it is very simple, easy and clear on how it should be used and if it is clearly explained, then people will see the benefits.  What is there to hate ?

**Comment by abrookbanks** [comment]
Having to clear cache before the front end updates manually. It will feel like a step backwards. Button position and education is key!

**Comment by abrookbanks** [comment]
I think this works really well already.

**Comment by havenswift-hosting** [comment]
Is this still a work in progress ?  I dont think the Clear Cache button is obvious enough and it should only be displayed if cache actually needs to be cleared.  Removing all those places where cache is automatically cleared is great though !

**Comment by abrookbanks** [comment]
> should only be displayed if cache actually needs to be cleared

I'm not sure it can be known aside from the obvious tables. Category, inventory ..

**Comment by abrookbanks** [comment]
* No changes made
![screen shot 2017-10-09 at 09 36 31](https://user-images.githubusercontent.com/4520803/31330239-70d7fe66-acd5-11e7-9206-fe381b26337b.png)

* Changes made
![screen shot 2017-10-09 at 09 36 46](https://user-images.githubusercontent.com/4520803/31330242-7320c54a-acd5-11e7-979b-45b15db04826.png)

**Comment by abrookbanks** [comment]
Really happy with this now but keen for more feedback.

**Comment by Dirty-Butter** [comment]
I would like it for sure!

**Comment by abrookbanks** [comment]
yes this is coming very soon

**Comment by abrookbanks** [comment]
This is working very well and I am happy. Needs real world testing.

**Comment by Dirty-Butter** [comment]
Well, this may or may not be related. I installed the latest 2.0 on dirtybutter.com/trial over the last commit of v2.0 I had tried, BUT no matter what I did I could not get the storefront to use proper css or possibly js. Whatever it was it looked like a narrow almost phone view. I used maintenance to clear all caches, used the clear cache button in Admin. I rebooted, cleared over and over with no success. I compared all files from the v2.0 download with the online version - all OK.
Today, I deleted ALL my browser temp files (My FireFox is set to automatically clear all browsing data on close so I hadn't suspected that) and manually deleted all temp files in trial. FINALLY I have a proper view of the storefront.
Like I said, it may have been a fluke - but since the way cache works has been changed recently - thought I'd better report it.
